sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -193,6 +193,15 @@ public ComponentDto selectOrFailByKey(DbSession session, String key) {
     return Optional.fromNullable(mapper(session).selectByKey(key));
   }
 
+  /**
+   * Select all Components in DB with the specified key.
+   *
+   * Used by Developer Cockpit
+   */
+  public List<ComponentDto> selectByKeyWithDuplicates(DbSession session, String key) {
```

Comment:
In fact this query already exists : selectComponentsHavingSameKeyOrderedById (used to remove possible duplicated keys when creating a project on MySQL)